### PR TITLE
Ignore error when retrieving changed PR review files

### LINKED ISSUE #21392: Server error on user-specific PR diff and a pruned commit
### Description

This is an edge case happening with the following conditions:
* an open PR with an assigned reviewer
* the reviewer interacted with the PR, Gitea stores user details about the interaction linked to a particular commit
* the author force-pushed to the PR
* git GC has run, causing a commit from the time of the interaction to be pruned

Observed behavior:
* the reviewer who interacted with the PR can't open PR Files Changed tab
* all other users can open it fine

Steps to replicate:
* In terminal
    * Run Gitea with basic docker-compose from https://docs.gitea.io/en-us/install-with-docker/#basics
* In browser
    * Install Gitea with SQLite database, admin user `author`
* In terminal
    * Shutdown Gitea
    * Configure git GC to prune unreachable objects immediately, append to `gitea/conf/app.ini`:
    ```
    [git]
    GC_ARGS = "--prune=now"
    ```
    * Run Gitea again
* In browser, logged as `author`
    * Add SSH key
    * Create repo `repo` with initialized `main` branch, user `reviewer`, add him to `repo` collaborators
* In terminal
    * Clone `repo`
    * Checkout branch `feature`, edit a file, commit, push
* In browser, logged as `author`
    * Create a PR for `feature` branch, assign `reviewer` as reviewer 
* In browser, logged as `reviewer`
    * Open the PR Files Changed tab
    * Mark the changed file as Viewed
* In terminal
    * Edit the file, amend the first commit, force-push
* In browser, logged as `author`
    * Site Administration > Maintenance Operations > Garbage collect all repositories > Run
    * Confirm that the first commit was pruned - opening it returns 404, this is expected
* In browser, logged as `reviewer`
    * Open the PR, Files Changed tab - opening returns 500
    * The following error is logged:
    ```
    gitea  | 2022/10/07 10:15:40 ...ers/web/repo/pull.go:715:ViewPullFiles() [E] [633fe02b] SyncAndGetUserSpecificDiff: exit status 128 - fatal: Invalid revision range 722a5b2dcdb1560ecfa0306c2492d1c5c7fd2a4e..47703d0b1e66ddf52827ff9596dc80f4de860089
    gitea  | 	 - fatal: Invalid revision range 722a5b2dcdb1560ecfa0306c2492d1c5c7fd2a4e..47703d0b1e66ddf52827ff9596dc80f4de860089
    ```
    * `722a5b…` is the pruned commit, `47703d…` is the amended commit

Expected behavior:
* PR Files Changed tab can be opened by anyone

Workaround:
* close the PR and reopen as a new PR

### Gitea Version

1.17.2

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

https://gist.github.com/zakjan/3e3c3574ac9df73d2e2bf0e717cd685d

### Screenshots

Screenshot 2022-10-10 at 9 47 23

### Git Version

2.36.2

### Operating System

macOS

### How are you running Gitea?

Docker

### Database

SQLite

### PR DESCRIPTION
**Pull request by delvh**
When a PR reviewer reviewed a file on a commit that was later gc'ed, they would always get a `500` response from then on when loading the PR.
This PR simply ignores that error and instead marks all files as unchanged.
This approach was chosen as the only feasible option without diving into **a lot** of error handling.

Fixes #21392

### CODE REVIEW

File: services/gitdiff/gitdiff.go [hunk 1]
```diff
@@ -1235,8 +1235,14 @@ func SyncAndGetUserSpecificDiff(ctx context.Context, userID int64, pull *issues_
...
 
 	changedFiles, err := gitRepo.GetFilesChangedBetween(review.CommitSHA, latestCommit)
+	// There are way too many possible errors.
+	// Examples are various git errors such as the commit the review was based on was gc'ed and hence doesn't exist anymore as well as unrecoverable errors where we should serve a 500 response
+	// Due to the current architecture and physical limitation of needing to compare explicit error messages, we can only choose one approach without the code getting ugly
+	// For SOME of the errors such as the gc'ed commit, it would be best to mark all files as changed
+	// But as that does not work for all potential errors, we simply mark all files as unchanged and drop the error which always works, even if not as good as possible
 	if err != nil {
-		return diff, err
+		changedFiles = []string{}
```
Comment (delvh) [inline services/gitdiff/gitdiff.go]: Alternatively,
```suggestion
		changedFiles = nil
```
or even
```suggestion
```
would be possible. Which approach is preferred?
I chose the current one due to being a little bit more explicit about its intention

**Review by lunny — APPROVED** [review]

File: services/gitdiff/gitdiff.go (hunk 1 above)
Comment (wxiaoguang) [reply services/gitdiff/gitdiff.go]: `changedFiles = nil` is clearer to me. 

But  I  think you can also just leave the `changedFiles` as it is and no need to change it.

**Review by wxiaoguang — APPROVED** [review]

**Comment by wxiaoguang** [comment]
I'd like to wait for @delvh 's feedback for comment https://github.com/go-gitea/gitea/pull/21487#discussion_r998891667 before merge.

File: services/gitdiff/gitdiff.go (hunk 1 above)
Comment (delvh) [reply services/gitdiff/gitdiff.go]: I'm fine with all approaches.
That's why I offered all three approaches.

> But I think you can also just leave the changedFiles as it is and no need to change it.

Do you mean
```diff
-		changedFiles = []string{}
```
or as it is right now?

Comment (wxiaoguang) [reply services/gitdiff/gitdiff.go]: Yup, if you could agree, we can take it to try

**Review by 6543 — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

**Pull request merged**